scipy/spatial/_spherical_voronoi.py

```diff
@@ -161,20 +149,60 @@ def __init__(self, points, radius=1, center=(0, 0, 0), threshold=1e-06):
         self.points = points
         self.radius = radius
         self.center = np.array(center)
+        self.threshold = threshold
 
-        if cKDTree(self.points).query_pairs(threshold * self.radius):
+        if cKDTree(self.points).query_pairs(self.threshold * self.radius):
             raise ValueError("Duplicate generators present.")
 
-        max_discrepancy = sphere_check(self.points,
-                                       self.radius,
-                                       self.center)
-        if max_discrepancy >= threshold * self.radius:
+        radii = np.linalg.norm(self.points - self.center, axis=1)
+        max_discrepancy = np.abs(radii - self.radius).max()
+        if max_discrepancy >= self.threshold * self.radius:
             raise ValueError("Radius inconsistent with generators.")
         self.vertices = None
         self.regions = None
         self._tri = None
         self._calc_vertices_regions()
 
+    def _handle_geodesic_input(self):
+
+        # center the points
+        centered = self.points - self.center
+
+        # calculate an orthogonal transformation using SVD
+        _, _, vh = np.linalg.svd(centered)
+
+        # calculate the north and south poles in this basis
+        poles = [[0, 0, self.radius], [0, 0, -self.radius]] @ vh
+
+        # project points into inverse basis (such that z-components are zero)
+        circle = centered @ vh.T[:, :2]
+
+        # simplicial neighbors are adjacent on the circle
+        angles = np.arctan2(circle[:, 1], circle[:, 0])
+        indices = np.argsort(angles)
+
+        # Voronoi vertices lie halfway between neighboring pairs
+        vertices = centered[indices] + centered[np.roll(indices, 1)]
+        vertices /= np.linalg.norm(vertices, axis=1)[:, np.newaxis]
+        vertices *= self.radius
+
+        # north and south poles are also Voronoi vertices
+        vertices = np.concatenate((vertices, poles))
+
+        # each region contains two vertices from the plane and the north and
+        # south poles
+        invf = np.argsort(indices)
+        invb = np.argsort(np.roll(indices, 1))
+
+        n = len(self.points)
+        regions = np.vstack([invf,                           # forward neigh.
+                             [len(self.points)] * n,         # north pole
+                             invb,                           # backward neigh.
+                             [len(self.points) + 1] * n]).T  # south pole
```

Comment:
we could just use `n` here and above I think,  since `n = len(self.points)`, but that can be cleaned up later if it matters.

Reply:
That was silly.  I have fixed it in #10692.